### PR TITLE
Feature/summarised dropdown fixes

### PR DESCRIPTION
**Pull request by kingy04**
Filters out an array with a empty value so the placeholder / placeholder summary display when an item is added and then removed

Allows the dropdown to resize to its parent container correctly, also uses an ellipsis if the parent goes to small for the content to prevent text overflow

Related Jiras: https://croudcontrol.atlassian.net/browse/CC-1376, https://croudcontrol.atlassian.net/browse/CC-1374

### CODE REVIEW

**Comment by paul-watkins-croud** [comment]
@kingy04 @BrockReece Shouldn't the data be sanitized before it comes into the component? There is nothing technically incorrect about anarray with an empty value - semnatically incorrect, but should the component to be opinionated about the data that's passed in?

To me the responsibility should lie with the consuming component to ensure that data is correct going into the dropdown as that is where the business logic is determined.

**Comment by BrockReece** [comment]
If we normalised this component only to deal with arrays by restricting the prop type and ensuring that is emits properly structured arrays wouldn't this fix things?

From what I can tell, this looks like it is curing the symptom and not the cause which is that the consuming component is probably attempting to split an empty string which gives `[""]` so where as I agree with what @paul-watkins-croud is saying, we are making things unnecessarily harder for the consuming component that has to deal with a mixture of arrays and strings when instead we could normalise in the styleguide and get some test coverage to match.

Just off the top of my head, something like this might help https://jsfiddle.net/tcgunoyr/5/

**Comment by kingy04** [comment]
@BrockReece, so i understand, we should set the prop to a string and emit an array? that is the change i have made.. can the tests wait due to the pending deadline?

File: src/components/shared/misc/SummarisedMultiSelector.vue [hunk 1]
```diff
@@ -159,7 +159,7 @@
                 },
 
                 set(val) {
-                    this.$emit('input', val)
+                    this.getArray(val)
```
Comment (BrockReece) [inline src/components/shared/misc/SummarisedMultiSelector.vue]: Shouldn't it still be emitting something like this? 
```js
this.$emit('input', this.getArray(val))
```

**Comment by BrockReece** [comment]
@kingy04 

No, we should set the prop to an array and emit an array. Then the parent can pass down an array (which is the usual structure returned in our API) and emit a new array when the value changes?

**Comment by kingy04** [comment]
I forgot to emit an array on the dropdown selected event too, apart from that this tested ok.

**Pull request merged**